### PR TITLE
Blog app - creating data model

### PR DESCRIPTION
**Pull request by Hamayounali**
## Created Models
1. Created user model.
2. Created post model.
3. Created a comment model.
4. Created like model.

### CODE REVIEW

File: db/migrate/20221026115832_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.integer :likes_counter
+      t.integer :comments_counter
+      t.references :user, null: false, foreign_key: true
```
Comment (Davidosky007) [inline db/migrate/20221026115832_create_posts.rb]: * Kindly Be sure to reference the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) from the sneak peek if you have any questions about the table or column names or data relationships. 
* As currently the foreign key `user` should be `author`  as shown in the `ERD` diagram [You can refer to the project requirements here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_1.md#:~:text=Be%20sure%20to%20reference%20the%20ERD%20diagram%20from%20the%20sneak%20peek%20if%20you%20have%20any%20questions%20about%20table%20or%20column%20names%20or%20data%20relationships.)
image

File: db/migrate/20221026120232_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :user, null: false, foreign_key: true
```
Comment (Davidosky007) [inline db/migrate/20221026120232_create_likes.rb]: * As mentioned above ⬆️ kindly stick to the `ERD` diagram for creating your foreign keys

File: db/migrate/20221026120534_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.references :user, null: false, foreign_key: true
```
Comment (Davidosky007) [inline db/migrate/20221026120534_create_comments.rb]: * As mentioned above ⬆️ kindly stick to the `ERD` diagram for creating your foreign keys

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Hamayounali 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Professional looking PR description 👍 
* Proper use of Gitflow 👍  

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20221026120232_create_likes.rb (hunk 2 above)
Comment (Hamayounali) [reply db/migrate/20221026120232_create_likes.rb]: Hi, @Davidosky007, it references the table user and it is the table name. Can I change the table name from user to author? according to the diagram, the table name should be user.

Comment (Davidosky007) [reply db/migrate/20221026120232_create_likes.rb]: > Hi, @Davidosky007, it references the table user and it is the table name. Can I change the table name from user to author? according to the diagram, the table name should be user.

Yes @Hamayounali but you don't have to change the table name you can use a Classname to achieve this ( ie  you can write something like this ` t.references :author, foreign_key: { to_table: 'users' })  this is just telling rails to treat classname `author` as user [You can readup this blog by our own student on classnaming ](https://www.microverse.org/blog/how-to-use-associations-and-class-names-in-ruby-on-rails)

Comment (rloterh) [reply db/migrate/20221026120232_create_likes.rb]: well resolved! ✔️

File: db/migrate/20221026120534_create_comments.rb (hunk 3 above)
Comment (rloterh) [reply db/migrate/20221026120534_create_comments.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Hamayoun,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Data model properly implemented

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Hamayounali** [comment]
Thank you @rloterh.

**Pull request merged**